src/jit/valuenum.cpp

```diff
@@ -1188,7 +1265,7 @@ ValueNum ValueNumStore::VNForMapSelectWork(ValueNumKind vnk, var_types typ, Valu
     VNDefFunc2Arg fstruct(VNF_MapSelect, arg0VN, arg1VN);
     if (GetVNFunc2Map()->Lookup(fstruct, &res))
     {
-        return res;
+        goto Return;
```

Comment:
In general we try to avoid using goto statements.
There already exist a TailCall label and got in this method.
I believe that you can replave the TailCall label with